### PR TITLE
Fixed #1370 : Added support for openload.co

### LINKED ISSUE #1370: openload.co support
https://openload.co/api

### PR DESCRIPTION
**Pull request by Bluecodelf**
I saw that the openload.co support was open for quite a long time, so I tried my hand at it. Every test case is working perfectly, and it has support for API authentication + folders.

### CODE REVIEW

File: ShareX.UploadersLib/app.config [hunk 1]
```diff
@@ -16,5 +16,18 @@ that is bundled with the nuget package under the tools folder.
 
 		<add key="AWSProfileName" value="" />
 -->
+    <add key="ClientSettingsProvider.ServiceUri" value="" />
   </appSettings>
+  <system.web>
+    <membership defaultProvider="ClientAuthenticationMembershipProvider">
```
Comment (Jaex) [inline ShareX.UploadersLib/app.config]: What are these?

**Comment by Bluecodelf** [comment]
Woops, missed that!

I removed the app.config file from my commit, everything still works perfectly. Visual Studio must have added those automatically, as I don't use this package at all.

File: ShareX.UploadersLib/ShareX.UploadersLib.csproj [hunk 2]
```diff
@@ -102,6 +102,7 @@
       <HintPath>..\Lib\Security.Cryptography.dll</HintPath>
     </Reference>
     <Reference Include="System" />
+    <Reference Include="System.Configuration" />
```
Comment (Jaex) [inline ShareX.UploadersLib/ShareX.UploadersLib.csproj]: Why there is two reference include here?

**Comment by Bluecodelf** [comment]
Indeed, my bad. I actually used the System.Configuration package for storing the folder nodes for the tree view into the configuration form at first, but that was unnecessary.

Amended my commit, should be good now :-)

**Comment by Jaex** [comment]
Are you doing force push or what, how you removing those without commit? This way you make me check whole diff every time instead of little commit.

**Comment by Bluecodelf** [comment]
Yes, I undo my commit to avoid polluting the commit list with mistakes in said commit. It makes undoing specific commits easier if needed, in my opinion. However I can see why it makes code reviewing more annoying, I can push further changes as new commits, if you want to.

Of course I would make a new commit if that changed the code directly, but this is usually something I do when I remove useless lines from build or project files from the last commit, when I spot them.

**Comment by Jaex** [comment]
I was not updated uploaders related stuff long days how you have so many conflicts.

**Comment by Bluecodelf** [comment]
That I do not know. Seems to be the added `<comment>@Invariant</comment>` lines from commit df5da6871b64eff1be84e3bb1b3ec7306d2b2b51 from nine days ago, in the file UploadersConfigForm.resx.

I guess I can try to merge them locally and redo a commit ?

**Comment by Jaex** [comment]
Those invariants not cause conflict.

**Comment by Bluecodelf** [comment]
Oh man, indeed. That's what I get for working on a old-ish fork, I guess. Git confuses me, sometimes, sorry. I'll make a fresh fork and re-implement my changes from here, I guess, and I'll get back to you in a sec!

**Pull request closed without merging**